packages/utils/src/Pending.ts

```diff
@@ -1,23 +1,33 @@
 import PromiseUtils from './PromiseUtils';
 
+interface CancelablePromise<T> extends Promise<T> {
+  cancel: () => void;
+}
+
 /**
  * Helper class for managing pending promises.
  * IDS-6391 When cancel is called, it calls cancel on ALL promises, including ones that have already resolved.
  * This behaviour may change when IDS-6806 is implemented.
  */
 class Pending {
-  pending = [];
+  pending: Array<CancelablePromise<unknown>> = [];
 
-  resolved = [];
+  resolved: Array<CancelablePromise<unknown>> = [];
 
   /**
    * Begins tracking a promise. After the promise has resolved, it's removed from tracking.
-   * @param {any} item Item to track.
-   * @param {function} cleanup The cleanup function to use when the promise is cancelled
+   * @param item Item to track.
+   * @param cleanup The cleanup function to use when the promise is cancelled
    * @returns CancelablePromise Returns a cancelable promise.
    */
-  add(item, cleanup = null) {
-    const promise = PromiseUtils.makeCancelable(item, cleanup);
+  add(
+    item: unknown,
+    cleanup: (val: unknown) => void | null
+  ): CancelablePromise<unknown> {
```

Comment:
```suggestion
  add<T>(
    item: Promise<T> | T,
    cleanup: (val: T) => void | null
  ): CancelablePromise<T> {
```

While unknown works, templating it will enforce the cleanup function uses the same param type as the item